pkg/pac/pac.go

```diff
@@ -226,25 +280,90 @@ func createBranch(client *gitlab.Client, projectID int, branchName string) error
 	return err
 }
 
-func createCommit(client *gitlab.Client, projectID int, branchName string, commitMessage string) error {
-	actionValue := gitlab.FileActionValue("create")
-	filePath := "preview.md"
-	content := "Preview changes for the new feature"
-
-	actions := []*gitlab.CommitActionOptions{{
-		Action:   &actionValue,
-		FilePath: &filePath,
-		Content:  &content,
-	}}
-	commitOptions := gitlab.CreateCommitOptions{
-		Branch:        &branchName,
+func createPacGenerateOpts() *pacgenerate.Opts {
+	// Initialize PAC generate options
+	opts := pacgenerate.MakeOpts()
+
+	// Set Event information
+	opts.Event = &info.Event{
+		EventType:  "pull_request",
+		BaseBranch: mainBranch,
+	}
+	// Set Project URL and Branch name to GitInfo
+	// ProjectURL is used as PipelineRun name with suffix
+	opts.GitInfo = &git.Info{
+		URL:    store.GetScenarioData("PROJECT_URL"),
+		Branch: mainBranch,
+	}
+	// Initialize I/O streams
+	var outputBuffer bytes.Buffer
+	opts.IOStreams = &cli.IOStreams{
+		Out:    &outputBuffer,
+		ErrOut: os.Stderr,
+		In:     os.Stdin,
+	}
+	// Specify the FileName of the pipelinerun yaml
+	opts.FileName = filePath
+
+	return opts
+}
+
+// Generate sample PipelineRun, pull-request.yaml
+func generatePipelineRun() error {
+	opts := createPacGenerateOpts()
+
+	err := pacgenerate.Generate(opts, true)
+	if err != nil {
+		return fmt.Errorf("failed to generate PipelineRun: %v", err)
+	}
+
+	return nil
+}
+
+// Validate generated yaml file from pac generate cmd
+func validateYAML(yamlContent []byte) error {
+	var content map[string]interface{}
+
+	err := yaml.Unmarshal(yamlContent, &content)
+	if err != nil {
+		return fmt.Errorf("invalid YAML format: %v", err)
+	}
+
+	return nil
+}
+
+func CreateCommit(client *gitlab.Client, projectID int, branch, commitMessage, fileDesPath string) error {
```

Comment:
IMHO this could be a private function (starting with lower case `c`) - it's not generic enough to be public